Docs: add note on matchExpressions for cnp and ccnp

darox opened the pull request:
This addition explains how a single matchExpressions works i.e. logical AND. On top of that it shows
how to achieve a logical OR by having multiple
matchExpressions.

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.

darox committed:
Docs: add note on matchExpressions for cnp and ccnp

This addition explains how a single matchExpressions
works i.e. logical AND. On top of that it shows
how to achieve a logical OR by having multiple
matchExpressions.

Signed-off-by: darox <maderdario@gmail.com>

qmonnet commented on Documentation/security/policy/kubernetes.rst:
```diff
@@ -48,6 +48,47 @@ Using namespace-specific information like
 will be ignored in `CiliumNetworkPolicy` resources.
 
+When using ``matchExpressions`` in a `CiliumNetworkPolicy` or 
+`CiliumClusterwideNetworkPolicy` the list values are
```
I'd rather have the references explicitly required to avoid any confusion:

```suggestion
When using ``matchExpressions`` in a :ref:`CiliumNetworkPolicy` or a
:ref:`CiliumClusterwideNetworkPolicy`, the list values are
```

(Feel free to add the missing one to the previous paragraph, as well.)

qmonnet commented on Documentation/security/policy/kubernetes.rst:
```diff
@@ -48,6 +48,47 @@ Using namespace-specific information like
+     .. group-tab:: k8s YAML
+
+        .. literalinclude:: ../../../examples/policies/l3/match-expressions/or-statement.yaml
+     .. group-tab:: JSON
```
```suggestion
        .. literalinclude:: ../../../examples/policies/l3/match-expressions/or-statement.yaml

     .. group-tab:: JSON
```

qmonnet commented on Documentation/security/policy/kubernetes.rst:
```diff
@@ -48,6 +48,47 @@ Using namespace-specific information like
+     .. group-tab:: k8s YAML
+
+        .. literalinclude:: ../../../examples/policies/l3/match-expressions/and-statement.yaml
+     .. group-tab:: JSON
```
```suggestion
        .. literalinclude:: ../../../examples/policies/l3/match-expressions/and-statement.yaml

     .. group-tab:: JSON
```

qmonnet commented on examples/policies/l3/match-expressions/and-statement.json:
```diff
@@ -0,0 +1,35 @@
+        }
+      ]
+    }
+  ]
```
Missing end-of-line character at the end of the file.

qmonnet commented on examples/policies/l3/match-expressions/or-statement.json:
```diff
@@ -0,0 +1,39 @@
+      }
+    ]
+  }
+]
```
Missing end-of-line character at the end of the file.

qmonnet reviewed (changes requested):
Some small nits.

I don't really like the structure of this subsection (“Known Pitfalls”), we have a list of pitfalls and then a list of examples but it's hard to visually relate each example to the related pitfall. Have you considered reworking it to have one pitfall followed by the relevant example, then the next followed by its relevant example, etc.?

(Please also get rid of the merge commit in your PR.)

darox committed:
Docs: restructure pitfalls section for policies

This commit moves pitfalls and examples for them
closer together, in order to make it more readable.

Signed-off-by: darox <maderdario@gmail.com>

qmonnet reviewed (commented):
Yes that's it, it looks great, thank you!

qmonnet reviewed (approved):
Awesome, thanks a lot!
